ee/desktop/runner/monitor_server.go

```diff
@@ -0,0 +1,54 @@
+package runner
+
+import (
+	"context"
+	"fmt"
+	"net"
+	"net/http"
+
+	"github.com/kolide/kit/ulid"
+)
+
+type monitorServer struct {
+	server   *http.Server
+	listener net.Listener
+	mux      *http.ServeMux
+}
+
+func newMonitorServer() (*monitorServer, error) {
+	listener, err := net.Listen("tcp", "localhost:0")
```

Comment:
Maybe we should use a socket? I have a feeling eventually someone will want to know why we are always listening on a random ports

Reply:
As you point out, we've seen weird socket issues. I say we try this, it's not that uncommon